server/repository/src/db_diesel/store_tag_row.rs

```diff
@@ -0,0 +1,248 @@
+use super::{store_tag_row::store_tag::dsl as store_tag_dsl, StorageConnection};
+
+use crate::repository_error::RepositoryError;
+
+use diesel::prelude::*;
+
+table! {
+    store_tag (id) {
+        id -> Text,
+        store_id -> Text,
+        tag_name -> Text,
+    }
+}
+
+#[derive(Clone, Queryable, Insertable, Debug, PartialEq, Eq, AsChangeset, Default)]
+#[table_name = "store_tag"]
+pub struct StoreTagRow {
+    pub id: String,
+    pub store_id: String,
+    pub tag_name: String,
+}
+
+// StoreTags are used to upsert/replace a set of tags for a store
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct StoreTags {
+    pub store_id: String,
+    pub tags: Vec<String>,
+}
+
+pub struct StoreTagRowRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+impl<'a> StoreTagRowRepository<'a> {
+    pub fn new(connection: &'a StorageConnection) -> Self {
+        StoreTagRowRepository { connection }
+    }
+
+    pub fn insert_one(&self, store_tag_row: &StoreTagRow) -> Result<(), RepositoryError> {
+        diesel::insert_into(store_tag_dsl::store_tag)
+            .values(store_tag_row)
+            .execute(&self.connection.connection)?;
+        Ok(())
+    }
+
+    pub fn find_one_by_id(&self, id: &str) -> Result<Option<StoreTagRow>, RepositoryError> {
+        let result = store_tag_dsl::store_tag
+            .filter(store_tag_dsl::id.eq(id))
+            .first(&self.connection.connection)
+            .optional()?;
+        Ok(result)
+    }
+
+    pub fn find_all_by_store_id(
+        &self,
+        store_id: &str,
+    ) -> Result<Vec<StoreTagRow>, RepositoryError> {
+        let result = store_tag_dsl::store_tag
+            .filter(store_tag_dsl::store_id.eq(store_id))
+            .load(&self.connection.connection)?;
+        Ok(result)
+    }
+
+    pub fn find_all_by_tag_name(
```

Comment:
nit: I really like just implementing what's currently needed, similar for find_one_by_store_id_and_tag_name